src/fillTank.test.js

```diff
@@ -1,11 +1,113 @@
 'use strict';
 
 describe('fillTank', () => {
-  // const { fillTank } = require('./fillTank');
+  const { fillTank } = require('./fillTank');
 
   it('should ', () => {
 
   });
 
-  // write tests here
+  it(`should not change fuelRemains if user has no money`, () => {
+    const customer1 = {
+      money: 0,
+      vehicle: {
+        maxTankCapacity: 40,
+        fuelRemains: 0,
+      },
+    };
+
+    fillTank(customer1, 10, 5);
+
+    expect(customer1.vehicle.fuelRemains).toBe(0);
+  });
+
+  it(`User.money should not change more`
+    + ` than the cost of filling up to maxTankCapacity`, () => {
+    const customer1 = {
+      money: 50,
+      vehicle: {
+        maxTankCapacity: 40,
+        fuelRemains: 20,
+      },
+    };
+
+    fillTank(customer1, 1, 40);
+
+    expect(customer1.money).toBe(30);
+  });
+
+  it(`should change fuelRemains to value that user can`
+  +` afford and fuelRemains <= maxTankCapacity`, () => {
+    const customer1 = {
+      money: 100, 
+      vehicle: {
+        maxTankCapacity: 40, 
+        fuelRemains: 0,
+      },
+    };
+  
+    fillTank(customer1, 1);
+  
+    expect(customer1.vehicle.fuelRemains).toBe(40); 
+    expect(customer1.money).toBe(60);
+  });
+  
+  it(`should change fuelRemains to amount of fuel that user can buy`, () => {
+    const customer1 = {
+      money: 20,
+      vehicle: {
+        maxTankCapacity: 40,
+        fuelRemains: 0,
+      },
+    };
+
+    fillTank(customer1, 1, 35);
+
+    expect(customer1.vehicle.fuelRemains).toBe(20);
+  });
+
+  it(`should not change fuelRemains if amount of fuels < 2`, () => {
```

Comment:
The test description mentions 'amount of fuels < 2', but it should be 'amount of fuel < 2' for grammatical correctness.